### PR TITLE
Fix #811, Updated SECURITY.md to fix error 404 for CodeSonar link 

### LINKED ISSUE #811: Security.md: Grammatech CodeSonar link is 404
**Checklist (Please check before submitting)**

* [x ] I reviewed the [Contributing Guide](https://github.com/nasa/cFS/blob/main/CONTRIBUTING.md).
* [ x] I performed a cursory search to see if the bug report is relevant, not redundant, nor in conflict with other tickets.

**Describe the bug**
Security.md contains a dead link describing CodeSonar

**To Reproduce**
1. Browse to https://github.com/nasa/cFS/security#codesonar
2. Click on the link
4. See error 404 from Grammatech

**Expected behavior**
Information about CodeSonar

**System observed on:**
 - Apple MacBookPro15,1
 - OS: MacOS Sonoma 14.7.1
 - Google Chrome Version 131.0.6778.109 (Official Build) (x86_64)

**Additional context**
N/A

**Reporter Info**
Douglas Held

### PR DESCRIPTION
**Pull request by vinsdragonis**
**Checklist (Please check before submitting)**

* [x] I reviewed the [Contributing Guide](https://github.com/nasa/cFS/blob/main/CONTRIBUTING.md).
* [x] I signed and emailed the appropriate [Contributor License Agreement](https://github.com/nasa/cFS/blob/main/CONTRIBUTING.md#contributor-license-agreement-cla) to GSFC-SoftwareRelease@mail.nasa.gov and copied cfs-program@lists.nasa.gov.

**Describe the contribution**
The URL has been updated in the markdown to redirect to the correct page for CodeSonar. The problem has to do with CodeSonar's page no longer being available on Grammatech and being moved to CodeSecure. It will no longer display a 404 error.

**Testing performed**
Steps taken to test the contribution:
1. Changes are reflected as expected in the markdown doc.
2. The URL in the markdown document now redirects to the appropriate page as seen.

**Expected behavior changes**
The URL should now redirects to the expected page.

**System(s) tested on**
 - Hardware: N/A
 - OS: N/A
 - Versions: N/A

**Additional context**
N/A

**Code contributions**
The cFS repository is provided to bundle the cFS Framework.  It is utilized for bundling submodules, continuous integration testing, and version management and does not contain any software.  Code contributions should be directed to the appropriate submodule.

**Contributor Info - All information REQUIRED for consideration of pull request**
Vineeth B V (Personal)

### CODE REVIEW

**Comment by vinsdragonis** [comment]
FIXES #811

**Review by RichLandau — APPROVED** [review]

**Comment by RichLandau** [comment]
Approved pending signed CLA

**Comment by vinsdragonis** [comment]
@RichLandau I have received approval for my CLA. I would kindly request you merge my PR.

Thanks.

**Pull request merged**
